db/migrate/20190513014444_create_books.rb

```diff
@@ -0,0 +1,10 @@
+class CreateBooks < ActiveRecord::Migration[5.2]
+  def change
+    create_table :books do |t|
+      t.string :name
+      t.string :isbn
```

Comment:
For fields that you know are required you can use the `null: false` option. This will add a DB constraint that prevents the data from being inserted into the table.

The reason we have both this and model validations is that model validations can be worked around.

EG:

```
book = Book.new
book.save # returns false because of validations
book.save(validate: false) # returns true and puts an invalid record in the DB
```

your migration will look like:

```ruby
class CreateBooks < ActiveRecord::Migration[5.2]
  def change
    create_table :books do |t|
      t.string :name, null: false
      t.string :isbn, null: false

      t.timestamps
    end
  end
end
```

If you make this change, you can re-run your migration with `rails db:migrate:redo`.

Also worth noting is that the timestamps are `null: false` by default (see the `schema.rb` below to confirm).